### PR TITLE
Fix Marchesa, Resolute Monarch's upkeep trigger

### LINKED ISSUE #12958: Back side of Invasion of Fiora triggers improperly
The flip side of [Invasion of Fiora](https://scryfall.com/card/mom/114/invasion-of-fiora-marchesa-resolute-monarch) (Marchesa, Resolute Monarch) appears to trigger the combat damage clause incorrectly. You draw a card and lose the life if you have taken damage, not if you haven't. It is supposed to be the other way around.

### PR DESCRIPTION
**Pull request by paasar**
Before fix it was triggering wrongly when player had been dealt damage since last turn.

Fix #12958

### CODE REVIEW

File: Mage.Tests/src/test/java/org/mage/test/cards/single/mom/InvasionOfFioraTest.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+        setChoice(playerA, true); // yes to cast Marchesa
+
+        checkPermanentCount("Marchesa on battlefield", 1, PhaseStep.END_TURN, playerA, marchesa, 1);
+
+        castSpell(2, PhaseStep.PRECOMBAT_MAIN, playerB, caress, bruiser);
+        waitStackResolved(2, PhaseStep.PRECOMBAT_MAIN);
+
+        // Researcher got +1/+1 counter from Caress's life gain
+        checkPT("Researcher PT", 2, PhaseStep.BEGIN_COMBAT, playerB, researcher, 3, 3);
```
Comment (paasar) [inline Mage.Tests/src/test/java/org/mage/test/cards/single/mom/InvasionOfFioraTest.java:64]: I wanted to check +1/+1 counters here but `CounterType.P1P1` kept returning 0.
Maybe I did it somehow wrong?

Comment (JayDi85) [reply Mage.Tests/src/test/java/org/mage/test/cards/single/mom/InvasionOfFioraTest.java:64]: > Whenever you gain life, put a +1/+1 counter on Blood Researcher

> you draw a card and you lose 1 life

Researcher need gain life, but marchesa give loose life - so no trigger?

Comment (xenohedron) [reply Mage.Tests/src/test/java/org/mage/test/cards/single/mom/InvasionOfFioraTest.java:64]: Gain life from Lich's Caress, not Marchesa, as the comment indicates.

Were you using `checkPermanentCounters`?

Comment (paasar) [reply Mage.Tests/src/test/java/org/mage/test/cards/single/mom/InvasionOfFioraTest.java]: Ah, yes. I guess I only tried `checkCardCounters`.

Now the test checks +1/+1 counters.

**Review by xenohedron — APPROVED** [review]
Thanks for the contribution! Always appreciate a clearly written test :)

**Pull request merged**